build.gradle

```diff
@@ -304,7 +304,7 @@ configure(javaCodeCheckedProjects) {
 def getGitVersion() {
     def describeStdOut = new ByteArrayOutputStream()
     exec {
-        commandLine 'git', 'describe', '--tags', '--always', '--dirty'
+        commandLine 'git', 'describe', '--dirty'
```

Comment:
Only pay attention to annotated tags now

Reply:
Release process document has been updated accordingly

Reply:
Not sure why this is needed.  This will break the versioning when not using tags.  So anyone using the driver and their own build process could now break.

Also how does this work for snapshots?

Reply:
I'm not clear how it will break the versioning.  Removing `--tags` just restricts the tags that we consider to ones that are annotated.  If we start annotating our tags this shouldn't change anything, except that it let's us add "lightweight" tags that are not considered for versioning.  My understanding is that annotating tags is a best practice that most MongoDB projects already follow.  Removing `--always` should have no effect as it's just a fallback in case no tags are found in the ancestry.  In practice I don't think could happen since every branch has some tag in its ancestry.

I think snapshots will continue to work.  This is just about the driver version that goes in the MongoDBVersion file, not the artifact name (which still is specified manually in build.gradle).  The snapshot will have the same version as before, something like `4.1.0-beta2-39-g3598ec6aa`.

Reply:
I can get it to fail locally: 
```
git describe --dirty
fatal: No annotated tags can describe 'eb3240d85e1b2522c75936d7160a24bfdc9dac0a'.
However, there were unannotated tags: try --tags.
mongo-kafka on  k124 [📝🤷‍] 
➜ git describe --dirty --tags
r1.2.0-10-geb3240d-dirty
```

Reply:
Right, this requires that we start annotating out tags.  In the Java driver itself, we haven't generally done this, but my intention is to start.

Reply:
The one weird thing about git describe on master, regardless of whether we annotate tags, is that often we don't tag any ancestor of master.  Consider the 4.1 release.  We branched 4.1.x prior to the r4.1.0-rc0 tag, so HEAD of master is not a descendant of it.  Perhaps whenever we branch we should tag the next commit to master (the one that bumps the version back to SNAPSHOT), e.g `r4.2.0-snapshot`

Reply:
So this would currently break publishing snapshots as we don't have any annotated tags.

I'm still unclear of the benefits of this change.  Does it actually change the output of the string if its annotated or not?
If it doesn't it will just potentially break users with forks for no benefit.  Also, if I fork the repo can I still build the project as I can now or will I have to pull in upstream tags as well?

Given `MongoDriverVersion` relies on this function I don't think it should be able to not produce a value and definitely should not suddenly start to error for users with forks of this repo.

Reply:
Yes, we have to push an annotated tag (though there actually is one, for some reason: whoever did the 3.4.0 tag annotated it).

It doesn't change the output.  It will just ignore any unannotated tags.

If you fork the repo, you do pick up all the tags. I wasn't sure, but I tried it here: https://github.com/jyemin/mongo-c-driver.  So a new fork should never be without tags. But you do have to manually synchronize new tags, so an existing fork would have to push --tags to get any new tags (annotated or not).  So removing `--always` is unlikely to break anyone since there should always be at least some tags in anyones fork. Removing `--tags` should also be ok, but only because of that stray 3.4.0 annotated tag that I mentioned earlier.

If you're still uncomfortable with the risks of this change, I will revert it. It's not essential to include it as part of this PR.

Reply:
Reverted this change